for v052, failed to generate the python sdk package on mac

it seems some cases failed, we should look into it

Here is the error message which may be the issue of OS of mac virtual machine.

```
/usr/local/lib/python3.9/site-packages/_pytest/assertion/rewrite.py:168: in exec_module
    exec(co, module.__dict__)
dbapi_test.py:20: in <module>
    from openmldb.dbapi import connect
/usr/local/lib/python3.9/site-packages/openmldb/__init__.py:16: in <module>
    from . import dbapi
/usr/local/lib/python3.9/site-packages/openmldb/dbapi/__init__.py:14: in <module>
    from .dbapi import *
/usr/local/lib/python3.9/site-packages/openmldb/dbapi/dbapi.py:22: in <module>
    from sdk import sdk as sdk_module
/usr/local/lib/python3.9/site-packages/openmldb/sdk/__init__.py:15: in <module>
    from .sdk import *
/usr/local/lib/python3.9/site-packages/openmldb/sdk/sdk.py:21: in <module>
    import native.sql_router_sdk as sql_router_sdk
/usr/local/lib/python3.9/site-packages/openmldb/native/__init__.py:15: in <module>
    from .sql_router_sdk import *
/usr/local/lib/python3.9/site-packages/openmldb/native/sql_router_sdk.py:13: in <module>
    from . import _sql_router_sdk
E   ImportError: dlopen(/usr/local/lib/python3.9/site-packages/openmldb/native/_sql_router_sdk.so, 2): initializer function 0x109f951a0 not in mapped image for /usr/local/lib/python3.9/site-packages/openmldb/native/_sql_router_sdk.so
```

--- a/.github/workflows/sdk.yml
+++ b/.github/workflows/sdk.yml
@@ -146,6 +146,7 @@ jobs:
     runs-on: macos-latest
     needs:
       - java-sdk
+    if: github.event_name == 'push'
     env:
       SQL_JAVASDK_ENABLE: ON
       OPENMLDB_BUILD_TARGET: 'cp_native_so openmldb'

--- a/.github/workflows/sdk.yml
+++ b/.github/workflows/sdk.yml
@@ -294,7 +294,7 @@ jobs:
           TWINE_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
 
   python-sdk-mac:
-    runs-on: macos-11
+    runs-on: macos-12
     env:
       SQL_PYSDK_ENABLE: ON
       OPENMLDB_BUILD_TARGET: 'cp_python_sdk_so openmldb'

--- a/.github/workflows/sdk.yml
+++ b/.github/workflows/sdk.yml
@@ -144,6 +144,7 @@ jobs:
     # except mvn deploy won't target all modules, just hybridse-native & openmldb-native
     # the job only run on tag push or manual workflow dispatch due to no test runs
     runs-on: macos-latest
+    if: github.event_name == 'push'
     needs:
       - java-sdk
     env:
@@ -294,7 +295,6 @@ jobs:
 
   python-sdk-mac:
     runs-on: macos-11
-    if: github.event_name == 'push'
     env:
       SQL_PYSDK_ENABLE: ON
       OPENMLDB_BUILD_TARGET: 'cp_python_sdk_so openmldb'

--- a/.github/workflows/sdk.yml
+++ b/.github/workflows/sdk.yml
@@ -146,7 +146,6 @@ jobs:
     runs-on: macos-latest
     needs:
       - java-sdk
-    if: github.event_name == 'push'
     env:
       SQL_JAVASDK_ENABLE: ON
       OPENMLDB_BUILD_TARGET: 'cp_native_so openmldb'
@@ -294,7 +293,7 @@ jobs:
           TWINE_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
 
   python-sdk-mac:
-    runs-on: macos-latest
+    runs-on: macos-11
     if: github.event_name == 'push'
     env:
       SQL_PYSDK_ENABLE: ON

--- a/.github/workflows/sdk.yml
+++ b/.github/workflows/sdk.yml
@@ -144,7 +144,6 @@ jobs:
     # except mvn deploy won't target all modules, just hybridse-native & openmldb-native
     # the job only run on tag push or manual workflow dispatch due to no test runs
     runs-on: macos-latest
-    if: github.event_name == 'push'
     needs:
       - java-sdk
     env:
@@ -295,6 +294,7 @@ jobs:
 
   python-sdk-mac:
     runs-on: macos-12
+    if: github.event_name == 'push'
     env:
       SQL_PYSDK_ENABLE: ON
       OPENMLDB_BUILD_TARGET: 'cp_python_sdk_so openmldb'

